[charts][docs] Revise the Axis doc (DX-38) (#19052)
mui/mui-x: docs/data/charts/axis/axis.md

### Automatic tick position

Use the `tickNumber` property to customize the number of ticks.

:::info

> This number does _not_ necessarily represent the exact number of ticks that will be displayed.

Reviewer: ```suggestion
This number does _not_ necessarily represent the exact number of ticks displayed.
```

> This is because D3 automatically places ticks to optimize for human readability, and it will round up or down from the provided `tickNumber` as needed to accomplish this.

Reviewer: ```suggestion
This is because D3 automatically places ticks to optimize for human readability, and it rounds up or down from the provided `tickNumber` as needed to accomplish this.
```

> For example, if you set `tickNumber=5` but there are only four years to display on the axis, the component will render four total ticks (one for each year) instead of trying to divide four years into five.

Reviewer: ```suggestion
For example, if you set `tickNumber=5` but there are only four years to display on the axis, the component renders four total ticks (one for each year) instead of trying to divide four years into five.
```

:::

To better control how the ticks render, you can also provide `tickMinStep` and `tickMaxStep`, which compute `tickNumber` so that the step between two ticks respects the minimum and maximum values.

In the demo below, the top axis has a `tickMinStep` of half a day, and the bottom axis has a `tickMinStep` of a full day.

{{"demo": "TickNumber.js"}}

### Fixed tick position

If you want more control over the tick position, you can use the `tickInterval` property.

> This property accepts an array of values that define where ticks will be placed.

Reviewer: ```suggestion
This property accepts an array of values that define exactly where ticks are placed.
```

Adding the "exactly" to mark the opposition with `tickNumber`

> For axes with the`'point'` scale type, the `tickInterval` property can be a filtering function of the type `(value, index) => boolean`.

Reviewer 2: ```suggestion
For axes with the `'point'` scale type, the `tickInterval` property can be a filtering function of the type `(value, index) => boolean`.
```
